test/e2e/kubeproxy.go

```diff
@@ -0,0 +1,447 @@
+			Name: nodePortServiceName,
+		},
+		Spec: api.ServiceSpec{
+			Type: api.ServiceTypeNodePort,
+			Ports: []api.ServicePort{
+				{Port: clusterHttpPort, Name: "http", Protocol: "TCP", NodePort: nodeHttpPort, TargetPort: util.NewIntOrStringFromInt(endpointHttpPort)},
+				{Port: clusterUdpPort, Name: "udp", Protocol: "UDP", NodePort: nodeUdpPort, TargetPort: util.NewIntOrStringFromInt(endpointUdpPort)},
+			},
+			Selector: selector,
+		},
+	}
+	config.nodePortService = config.f.createService(serviceSpec)
+}
+
+func (config *TestConfig) createLoadBalancerService(selector map[string]string) {
+	serviceSpec := &api.Service{
+		ObjectMeta: api.ObjectMeta{
+			Name: loadBalancerServiceName,
+		},
+		Spec: api.ServiceSpec{
+			Type: api.ServiceTypeLoadBalancer,
+			Ports: []api.ServicePort{
+				{Port: loadBalancerHttpPort, Name: "http", Protocol: "TCP", TargetPort: util.NewIntOrStringFromInt(endpointHttpPort)},
+			},
+			Selector: selector,
+		},
+	}
+	config.f.createService(serviceSpec)
+}
+
+func (config *TestConfig) waitForLoadBalancerIngressSetup() {
+	err := wait.Poll(5*time.Second, 45*time.Second, func() (bool, error) {
+		service, err := config.f.getServiceClient().Get(loadBalancerServiceName)
+		if err != nil {
+			return false, err
+		} else {
+			if len(service.Status.LoadBalancer.Ingress) > 0 {
+				return true, nil
+			} else {
+				return false, fmt.Errorf("Service LoadBalancer Ingress was not setup.")
+			}
+		}
+	})
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to setup Load Balancer Service. err:%v", err))
+	config.loadBalancerService, _ = config.f.getServiceClient().Get(loadBalancerServiceName)
+}
+
+func (config *TestConfig) createTestPod() {
+	testContainerPod := config.f.createTestPodSpec()
+	config.testContainerPod = config.f.createPod(testContainerPod)
+}
+
+func (f *Framework) createService(serviceSpec *api.Service) *api.Service {
+	_, err := f.getServiceClient().Create(serviceSpec)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create %s service: %v", serviceSpec.Name, err))
+
+	err = waitForService(f.Client, f.Namespace.Name, serviceSpec.Name, true, 5*time.Second, 45*time.Second)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("error while waiting for service:%s err: %v", serviceSpec.Name, err))
+
+	createdService, err := f.getServiceClient().Get(serviceSpec.Name)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create %s service: %v", serviceSpec.Name, err))
+
+	return createdService
+}
+
+func (f *Framework) cleanup() {
+	nsClient := f.getNamespaceClient()
+	nsList, err := nsClient.List(nil, nil)
+	if err == nil {
+		for _, ns := range nsList.Items {
+			if strings.Contains(ns.Name, f.BaseName) && ns.Name != f.Namespace.Name {
+				nsClient.Delete(ns.Name)
+			}
+		}
+	}
+}
+
+func (f *Framework) createNetProxyPods(podName string, selector map[string]string, nodeCount int) []*api.Pod {
+	//testContext.CloudConfig.NumNodes
+	pods := make([]*api.Pod, 0)
+
+	for i := 0; i < nodeCount; i++ {
```

Comment:
Why not a replication controller?  If there's a good reason it should be commented here